sdk/python/lib/pulumi/provider/experimental/ComponentProvider.md

```diff
@@ -0,0 +1,185 @@
+# Component Provider
+
+The `pulumi.provider.experimental` package allows writing a Pulumi provider in Python with minimal boilerplate, using type annotations to define the schema. The provider can be published to Git or referenced locally.
+
+## Building a Component Provider
+
+Create a new directory that will hold the provider implementation.
+
+First we need to let Pulumi's plugin system know which language runtime should be used to run the provider plugin. Create a a `PulumiPlugin.yaml` file with the contents:
+
+```yaml
+runtime: python
+```
+
+> [!NOTE]
+> Python component providers only support the pip toolchain with a virtual environment named `venv`. Don't add any other runtime options to the `PulumiPlugin.yaml` file.
+
+We also need to create a `requirements.txt` file to list the dependencies of the provider:
+
+```txt
+git+ssh://git@github.com/pulumi/pulumi.git@julienp/tutorial#subdirectory=sdk/python
+```
+
+TODO: update version to a pre-release with the experimental component provider merged
+
+
+The resources provided by the provider are are defined as Python classes that subclass `pulumi.ComponentResource`. Create a new file `my-component.py` to hold the component implementation:
+
+```python
+from typing import Optional
+
+import pulumi
+
+class MyComponent(pulumi.ComponentResource):
+    def __init__(self, name: str, opts: Optional[pulumi.ResourceOptions] = None):
+        super().__init__("my-provider:index:MyComponent", name, {}, opts)
+        self.register_outputs({})
+```
+
+> [!NOTE]
+> The type string `my-provider:index:MyComponent` follows the pattern `<package>:<module>:<type>`. The package is your provider's name, module must be `index`. The `type` must match the class name.
+
+Next, create a `__main__.py` file to host the provider:
+
+```python
+from pulumi.provider.experimental import Metadata, component_provider_host
+
+if __name__ == "__main__":
+    component_provider_host(
+        Metadata(name="my-provider", version="1.2.3", display_name="My Component Provider")
```

Comment:
Is there value in supplying `name` and `version` explicitly version deriving them from the package name and GH tag? What does `display_name` give? (I assume, a nice Registry page, eventually?)

Reply:
I think for now we need at least the name to be provided. The plugin is stored in `~/.pulumi/plugins/resource-github.com_julienp_cloudrun.git-v2.0.2/`, which is not really suitable for automatically determine a name.

Maybe we can strip `.git-*` from the end, split on `_` and take the last part?

Reply:
That would be an accurate way to get the name of the repository.